tests/functional/aws-node-sdk/test/multipleBackend/objectTagging.js

```diff
@@ -0,0 +1,221 @@
+const tagParams = { Bucket: bucket, Key: key };
+const awsDelParams = { Bucket: awsBucket, Key: key };
+const testBackends = ['mem', 'file', 'aws-test'];
+const tagString = 'key1=value1&key2=value2';
+const putTags = {
+    TagSet: [
+        {
+            Key: 'key1',
+            Value: 'value1',
+        },
+        {
+            Key: 'key2',
+            Value: 'value2',
+        },
+    ],
+};
+
+function getObject(backend, tagCheck, cb) {
+    s3.getObject({ Bucket: bucket, Key: key }, (err, res) => {
+        assert.equal(err, null);
+        assert.strictEqual(res.ETag, `"${correctMD5}"`);
+        assert.strictEqual(res.Metadata['scal-location-constraint'],
+            backend);
+        if (tagCheck) {
+            assert.strictEqual(res.TagCount, '2');
+        } else {
+            assert.strictEqual(res.TagCount, undefined);
+        }
+        if (backend === 'aws-test') {
+            setTimeout(() => {
+                awsS3.getObject({ Bucket: awsBucket, Key: key }, (err, res) => {
+                    assert.equal(err, null);
+                    assert.strictEqual(res.ETag, `"${correctMD5}"`);
+                    if (tagCheck) {
+                        assert.strictEqual(res.TagCount, '2');
+                    } else {
+                        assert.strictEqual(res.TagCount, undefined);
+                    }
+                    cb();
+                });
+            }, awsTimeout);
+        } else {
+            cb();
+        }
+    });
+}
+
+describeSkipIfNotMultiple('Object tagging with multiple backends',
+function testSuite() {
+    this.timeout(30000);
+    withV4(sigCfg => {
+        beforeEach(() => {
+            const awsConfig = getRealAwsConfig(awsLocation);
+            awsS3 = new AWS.S3(awsConfig);
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return s3.createBucketAsync({ Bucket: bucket })
+            .catch(err => {
+                process.stdout.write(`Error creating bucket: ${err}\n`);
+                throw err;
+            });
+        });
+
+        afterEach(() => {
+            process.stdout.write('Emptying bucket\n');
+            return bucketUtil.empty(bucket)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(bucket);
+            })
+            .catch(err => {
+                process.stdout.write(`Error in afterEach: ${err}\n`);
+                throw err;
+            });
+        });
+
+        describe('putObject with tags and putObjectTagging', () => {
+            testBackends.forEach(backend => {
+                it(`should put an object with tags to ${backend} backend`,
+                done => {
+                    const params = Object.assign({ Tagging: tagString, Metadata:
+                        { 'scal-location-constraint': backend } }, putParams);
+                    s3.putObject(params, err => {
```

Comment:
I assume this is not working for the aws backend since we are not pulling the dataStore properly on 0 byte objects? We should test with both 0 byte objects and objects with some data to make sure we are testing both code paths.